### ISSUE TITLE
Add JPA 3.0 support

### ISSUE THREAD

**Issue by fanste** [eclipse-dali/webtools.dali#6]
There is already a PR for this in the old bugtracker, but nothing has come to an end until now.

- https://bugs.eclipse.org/bugs/show_bug.cgi?id=579533
- https://git.eclipse.org/r/c/dali/webtools.dali/+/193026
- https://www.eclipse.org/forums/index.php?t=msg&th=1111243&goto=1854579&

Any change to get something updated here regarding newer JPA versions?

**Comment by lak-proddev** [comment]
@nitind Please think  about adding this to 2024 -06.

I believe we can include the basic JPA 3.0  support. (that provides the JPA 3.0 facet). Since many people are requesting for it.

Still, some work is pending for JPA 3.0 support due to package changes.
https://bugs.eclipse.org/bugs/show_bug.cgi?id=579533#c13

**_In JPA 3.0 the package name changed from "javax.persistence" to "jakarta.persistence".
We need to make these changes in code generation._** 

This is something users can manage manually for now.

**Comment by lak-proddev** [comment]
@nitind  Please merge it ASAP. I need it for testing with the libraries.

**Comment by nitind** [comment]
@lak-proddev done.

**Comment by lak-proddev** [comment]
> @lak-proddev done. 

Thank you very much.

**Comment by fanste** [comment]
Looks like the changes are already added to master. Thank you.

When might we expect a release available for eclipse IDE?

**Comment by lak-proddev** [comment]
> Looks like the changes are already added to master. Thank you.
> 
> When might we expect a release available for eclipse IDE?
It's been part of 2024-06 release. Please note, that it's just basic facet support.

**Comment by WolfgangHG** [comment]
The "New JPA entity" wizard still (using Eclipse 2024-12) still declares the old imports when facet "JPA 3.0" is selected:

```java
import javax.persistence.*;

@Entity
public class MyEntityBean implements Serializable {
```

![Image](https://github.com/user-attachments/assets/05e4ea30-8f96-4e7d-be28-a04045bd2b2d)

Is this the right place to report this or should I create a new one?

**Comment by lak-proddev** [comment]
> The "New JPA entity" wizard still (using Eclipse 2024-12) still declares the old imports when facet "JPA 3.0" is selected:
> import javax.persistence.*;
> Is this the right place to report this or should I create a new one?

Yes, but there is already one created for this issue. https://github.com/eclipse-dali/webtools.dali/issues/10

Due to time constraints and other priority issues, I am unable to spend time on it. I welcome the volunteers who are interested in fixing the issue.


> Is this support fully added? I can see several changes on master for JPA 3.0 and 3.1, but issues are still open and I don't know if something is still not working.
> 
No, we have just added the facet support.

**Comment by Emkas** [comment]
> No, we have just added the facet support.

OK. Thank you for fast response!

**Comment by lak-proddev** [comment]
> The "New JPA entity" wizard still (using Eclipse 2024-12) still declares the old imports when facet "JPA 3.0" is selected:
> Is this the right place to report this or should I create a new one?

We can fix it separately. I have created a new one for it.
Thanks @WolfgangHG 
https://github.com/eclipse-dali/webtools.dali/issues/13

**Comment by lak-proddev** [comment]
Added JPA 3.x support up to version 3.2. We will update incrementally where required.
